### PR TITLE
doc: add a document to specify the xDS bootstrap file format

### PR DESCRIPTION
**Pull request by easwars**

### CODE REVIEW

File: doc/grpc_xds_bootstrap_format.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+          "config": <JSON object containing config for the type>
+        }
+      ],
+
+      // A list of features supported by the server.  New values will
+      // be added over time.  For forward compatibility reasons, the
+      // client will ignore any entry in the list that it does not
+      // understand, regardless of type.
+      // 
+      // [Supported Server Features](#supported-server-features)
```
Comment (markdroth) [inline doc/grpc_xds_bootstrap_format.md:51]: Same here (link not working).

File: doc/grpc_xds_bootstrap_format.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      "sub_zone": <string>,
+    },
+
+    // Opaque metadata extending the node identifier.
+    "metadata": <JSON Object>,
+  }
+
+  // Map of supported certificate providers, keyed by the provider instance
+  // name.
+  // [Supported certificate providers](#supported-certificate-provider-instances)
```
Comment (markdroth) [inline doc/grpc_xds_bootstrap_format.md:79]: Same here (link not working).

**Review by markdroth — COMMENTED** [review]
Looks good overall!  Just a few minor comments.

File: doc/grpc_xds_bootstrap_format.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+The following sections describe the bootstrap file format, including links to
+gRFCs where support for appropriate fields was added.
+
+## File Format
+
+```
+{
+  // The xDS server to talk to.  The value is an ordered array of server
+  // configurations, to support failing over to a secondary xDS server if the
```
[the remark this answers is not in the document]
Comment (easwars) [reply doc/grpc_xds_bootstrap_format.md:24]: Done.

Markdown doesn't seem to support links from within backticks. So, I couldn't link to A71 from this comment. Shall I leave it as it stands currently, or do you think it makes sense to put the complete URL to A71 here?

File: doc/grpc_xds_bootstrap_format.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+  // primary is down.
+  "xds_servers": [
+    {
+
+      "server_uri": <string containing URI of xds server>,
```
[the remark this answers is not in the document]
Comment (easwars) [reply doc/grpc_xds_bootstrap_format.md:29]: Done. Thanks.

File: doc/grpc_xds_bootstrap_format.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+      // List of channel creds; client will stop at the first type it
+      // supports.  This field is required and must contain at least one
+      // channel creds type that the client supports.
+      //
+      // [Supported Channel Credentials](#supported-channel-credentials)
```
[the remark this answers is not in the document]
Comment (easwars) [reply doc/grpc_xds_bootstrap_format.md:35]: I was hoping that Github would do something smart here. But doesn't look like it. Let me know how this looks. Thanks.

File: doc/grpc_xds_bootstrap_format.md (hunk 1 above)
Comment (easwars) [reply doc/grpc_xds_bootstrap_format.md:51]: Done.

File: doc/grpc_xds_bootstrap_format.md (hunk 2 above)
Comment (easwars) [reply doc/grpc_xds_bootstrap_format.md:79]: Done.

File: doc/grpc_xds_bootstrap_format.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (markdroth) [reply doc/grpc_xds_bootstrap_format.md:24]: I suggest just adding a link to A71 in the table below that shows the gRFCs in which each field is specified.

[the remark this answers is not in the document]
Comment (easwars) [reply doc/grpc_xds_bootstrap_format.md:24]: Done

**Review by markdroth — APPROVED** [review]
Looks great!  Thanks for doing this!

**Pull request closed without merging**